src/fontra/views/editor/edit-tools-pen.js

```diff
@@ -417,6 +417,15 @@ function setupExistingAnchorPoint(context, path, point, shiftConstrain) {
 }
 
 function insertAnchorPoint(context, path, point, shiftConstrain) {
+  if (shiftConstrain) {
+    if (context.anchorIndex >= 1) {
+      // the path need at least 2 points
+      // to be able to shift constrain
+      const previousPoint = path.getPoint(context.anchorIndex - 1);
+      point = getHandle(point, previousPoint, shiftConstrain);
```

Comment:
- Hmm, we may want to rename `getHandle()`, as we're not dealing with a handle here. I see that the logic is correct.
- The `anchorIndex >= 1` is not what we need to check
- Check for `context.createContour`: we should only constrain when it is `false` (if we're creating a new contour, the inserted point is the first of a new contour, and we don't have a reference point)
- But then we have `context.contourIndex` and `context.contourPointIndex`, and you can get the reference point with `path.getContourPoint(context.contourIndex, context.contourPointIndex)`

Reply:
Or actually, let's not reuse `getHandle()` for this, as it redundantly checks for `constrain`, so we can use `shiftConstrain()` directly. `vector.roundVector()` is also already done.

Reply:
I get your points. Funny enough, but when we use the function shiftConstrain, we get in a conflict with the variable 'shiftConstrain'. We wanted to solve this in a different task. Should I fix it only within 'insertAnchorPoint' or should we do it for the while file for consistency?

Reply:
Please only fix it in `insertAnchorPoint()`.